Automattic/woocommerce-payments
# Don't process payment intent success webhooks for WCPay subscriptions related intents

Fixes #3198 

#### Changes proposed in this Pull Request

This PR fixes a bug introduced in https://github.com/Automattic/woocommerce-payments/pull/3118 whereby a payment intent is received without metadata containing the order ID. 

It also fixes a compatibility issue with WCPay subscriptions where WC Payments would try to locate an order linked to a payment intent, but because the intent was created on Stripe's end, not via the checkout, the intent doesn't metadata including the order ID. 

WC Payments handles payment intent webhooks differently (via the `invoice.paid` event) and so doesn't need to be handled by this function.  

### Requirements:
- WC Pay Subscription billing clocks via the WCPay Dev tools plugin.
- WC Pay Subscriptions enabled. 
- Webhook listening enabled.
     - WC Pay Server `npm run listen`

### To test:

1. Purchase a subscription product using WC Payments.
2. Go to the edit subscription screen.
3. Enable Dev Testing Mode on the right-hand side of the screen.
4. Process the invoice upcoming, invoice created, and invoice paid events. 
5. You should get the error mentioned above. 

-------------------


**Post merge**
